src/main/java/com/kickstarter/dropwizard/metrics/influxdb/InfluxDbMeasurement.java

```diff
@@ -147,34 +152,85 @@ public Builder putTag(final String key, final String value) {
     /**
      * Adds the given key-value pair to the fields map.
      *
-     * @throws IllegalArgumentException if any value is not a String or primitive.
+     * @throws IllegalArgumentException if any value is not a String, primitive,
+     *         or Collection of Strings and primitives.
      */
     public <T> Builder putField(final String key, final T value) {
-      if (value == null) {
-        return this;
+      if (value instanceof Collection<?>) {
+        final Collection collection = (Collection) value;
+        final String collString = validatedPrimitiveCollection(key, collection);
+        fields.put(key, collString);
+      } else if (value != null) {
+        final Optional<String> fieldString = validatedPrimitiveField(key, value);
+        fieldString.ifPresent(s -> fields.put(key, s));
+      }
+
+      return this;
+    }
+
+    /**
+     * Validates that the collection only contains Strings and primitives.
```

Comment:
if we include all primitives, should our list above include `Short`, `Byte`  and `Character` too?  (I don't know enough background to know if that's realistic in this lib 😬 ) 

PS on that note, I wonder if there's a builtin list of java primitives? 🤔